File: it/structures/csharp/enum-default/Test/EnumExplicit.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System.Runtime.Serialization;

namespace Test {
    /// Explicitly assigned strings
    [JsonConverter(typeof(StringEnumConverter))]
    public enum EnumExplicit {
        [EnumMember(Value = "foo")]
        A,
        [EnumMember(Value = "bar")]
        B
    }
}

File: it/structures/java/enum-default/test/EnumExplicit.java
package test;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Explicitly assigned strings
 */
public enum EnumExplicit {
    A("foo"),
    B("bar");

    String value;

    EnumExplicit(final String value) {
        this.value = value;
    }

    @JsonCreator
    public static EnumExplicit fromValue(final String value) {
        for (final EnumExplicit v : values()) {
            if (v.value.equals(value)) {
                return v;
            }
        }

        throw new IllegalArgumentException("value");
    }

    @JsonValue
    public String toValue() {
        return this.value;
    }
}
